.github/workflows/misc-sync-templates.yml

```diff
@@ -27,7 +27,66 @@ on:
         default: false
 
 jobs:
+  prepare-chain-spec-artifacts:
+    runs-on: ubuntu-latest
+    strategy:
+      matrix:
+        include:
+          - template: minimal
+            package_name: 'minimal-template-runtime'
+            runtime_path: './templates/minimal/runtime'
+            runtime_wasm_path: minimal-template-runtime/minimal_template_runtime.compact.compressed.wasm
+            relay_chain: 'dev'
+          - template: parachain
+            package_name: 'parachain-template-runtime'
+            runtime_path: './templates/parachain/runtime'
+            runtime_wasm_path: 'minimal-template-runtime/minimal_template_runtime.compact.compressed.wasm'
+            relay_chain: 'rococo-local'
+    steps:
+        - uses: actions/checkout@v4
+          with:
+            ref: "${{ github.event.inputs.stable_release_branch }}"
+
+        - name: Setup build environment
+          run: |
+            sudo apt-get update
+            sudo apt-get install -y protobuf-compiler
+            cargo install --git https://github.com/chevdor/srtool-cli --locked
+            cargo install --path substrate/bin/utils/chain-spec-builder --locked
+            srtool pull
+            
+        - name: Build runtime and generate chain spec
+          run: |
+            # Prepare directories
+            sudo mkdir -p ${{ matrix.runtime_path }}/target
+            sudo chmod -R 777 ${{ matrix.runtime_path }}/target
+            
+            # Build runtime
+            srtool build --package ${{ matrix.package_name }} --runtime-dir ${{ matrix.runtime_path }} --root
+            
+            # Generate chain spec
+            # Note that para-id is set to 1000 for both minimal/parachain templates. 
+            # `parachain-runtime` is hardcoded to use this parachain id. 
+            # `minimal` template isn't using it, but when started with Omni Node, this para id is required (any number can do it, so setting it to 1000 for convenience).
+            chain-spec-builder -c dev_chain_spec.json create \
+              --relay-chain "${{ matrix.relay_chain }}" \
+              --para-id 1000 \
+              --runtime "${{ matrix.runtime_path }}/target/srtool/release/wbuild/${{ matrix.runtime_wasm_path }}" \
+              named-preset development
+
+        - name: Prepare upload directory
+          run: |
+            mkdir -p artifacts-${{ matrix.template }}
+            cp dev_chain_spec.json artifacts-${{ matrix.package_name }}/dev_chain_spec.json
```

Comment:
Here you are using `matrix.package_name` instead of `matrix.template`, I think this is a mismatch.

Reply:
Thanks for the catch. I have addressed it